cmd/azmount/filemanager/azure.go

```diff
@@ -70,71 +87,95 @@ func AzureSetup(urlString string, urlPrivate bool, identity common.Identity) err
 	if err != nil {
-		return errors.Wrapf(err, "can't parse URL string")
+		return errors.Wrapf(err, "Can't parse URL string %s", urlString)
 	}
 
 	if urlPrivate {
 		// we use token credentials to access private azure blob storage the blob's
 		// url Host denotes the scope/audience for which we need to get a token
-		logrus.Infof("Using token credentials")
+		logrus.Info("Using token credentials to access private azure blob storage...")
 
 		var token common.TokenResponse
 		count := 0
+		logrus.Infof("Getting token for https://%s", u.Host)
 		for {
 			token, err = common.GetToken("https://"+u.Host, identity)
 
 			if err != nil {
-				logrus.Infof("can't obtain a token required for accessing private blobs. will retry in case the ACI identity sidecar is not running yet.")
+				logrus.Info("Can't obtain a token required for accessing private blobs. Will retry in case the ACI identity sidecar is not running yet...")
 				time.Sleep(3 * time.Second)
 				count++
 				if count == 20 {
-					return errors.Wrapf(err, "timeout of 60 seconds expired. could not obtained token")
+					return errors.Wrapf(err, "Timeout of 60 seconds expired. Could not obtain token")
 				}
 			} else {
-				logrus.Infof("token obtained: %s continuing", token.AccessToken)
+				logrus.Infof("Token obtained: %s \nContinuing...", token.AccessToken)
 				break
 			}
 		}
 
 		tokenCredential := azblob.NewTokenCredential(token.AccessToken, tokenRefresher)
+		logrus.Infof("Token credential created: %s", tokenCredential.Token())
 		fm.blobURL = azblob.NewPageBlobURL(*u, azblob.NewPipeline(tokenCredential, azblob.PipelineOptions{}))
+		logrus.Infof("Blob URL created: %s", fm.blobURL)
 	} else {
 		// we can use anonymous credentials to access public azure blob storage
-		logrus.Infof("Using anonymous credentials")
+		logrus.Info("Using anonymous credentials to access public azure blob storage...")
 
 		anonCredential := azblob.NewAnonymousCredential()
+		logrus.Infof("Anonymous credential created: %s", anonCredential)
 		fm.blobURL = azblob.NewPageBlobURL(*u, azblob.NewPipeline(anonCredential, azblob.PipelineOptions{}))
+		logrus.Infof("Blob URL created: %s", fm.blobURL)
 	}
 
 	// Use a never-expiring context
 	fm.ctx = context.Background()
-	logrus.Infof("Getting size of file...")
 
+	logrus.Info("Getting size of file...")
 	// Get file size
 	getMetadata, err := fm.blobURL.GetProperties(fm.ctx, azblob.BlobAccessConditions{},
 		azblob.ClientProvidedKeyOptions{})
 	if err != nil {
-		return errors.Wrapf(err, "can't get size")
+		return errors.Wrapf(err, "Can't get blob file size")
 	}
 	fm.contentLength = getMetadata.ContentLength()
-	logrus.Infof("Size: %d bytes", fm.contentLength)
+	logrus.Infof("Blob Size: %d bytes", fm.contentLength)
 
-	// Setup data downloader
+	// Setup data downloader and uploader
 	fm.downloadBlock = AzureDownloadBlock
+	fm.uploadBlock = AzureUploadBlock
+
+	return nil
+}
+
+func AzureUploadBlock(blockIndex int64, b []byte) (err error) {
```

Comment:
Does this belong in this PR? It doesn't look like logging changes.

Reply:
I branched from the R/W update so that I would have the changes I made there (I added some logs to debug with). I assumed the R/W PR would get merged before this.

Reply:
Rebased this PR to point to the R/W branch instead of main.  This is now moot.